api/src/routes/identity/history/get.ts

```diff
@@ -51,30 +56,74 @@ async function resolveHistory(
             config.setPermanode(permaNodeUrl);
         }
 
-        // Create a client instance to publish messages to the Tangle.
         const client = identity.Client.fromConfig(config);
 
-        const recepit = await client.resolveHistory(did);
-        const recepitObj = recepit.toJSON();
+        const receipt = await client.resolveHistory(did);
+        const receiptObj = receipt.toJSON();
 
         const integrationChainData = [];
 
-        for (const element of recepit.integrationChainData()) {
+        for (const element of receipt.integrationChainData()) {
             const integrationMessage = {
                 document: element.toJSON(),
+                messageId: element.toJSON().integrationMessageId
+            };
+            integrationChainData.push(integrationMessage);
+        }
+
+        const history = {
+            integrationChainData,
+            diffChainData: receiptObj.diffChainData,
+            diffChainSpam: receiptObj.diffChainSpam
+        };
+
+        return history;
+    } catch (e) {
+        return { error: typeof e === "string" ? e : e.toString() };
+    }
+}
+
+/**
+ * @param  {string} did DID to be resolved
```

Comment:
The format looks wrong here. If we are specifying types, we should also do that for other params.